lib/osv/cache.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Caching interface and implementations"""
+import datetime
+import typing
+
+import redis
+
+
+class Cache:
+  """Cache Interface"""
+
+  def get(self, key):
+    raise NotImplementedError
+
+  def set(self, key, value, ttl):
+    raise NotImplementedError
+
+
+class RedisCache(Cache):
```

Comment:
Can we add this implementation to the worker.py instead? That way, not all users of the "osv" library has to install the redis package.

Reply:
Moved to worker, though it seems like something else is also depending on redis, because the pipfile.lock file still has the redis dependency even though it's been removed from the pipfile

Reply:
pipenv graph shows `google-cloud-ndb` pulling in redis as a dependency

Reply:
Interesting. It's not ideal in the first place that `google-cloud-ndb` is included as a dependency for OSV, so let's not make this worse and keep the explicit redis dependency in the worker.